common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java

```diff
@@ -1170,11 +1173,29 @@ private void verifyTokenParametersAreSupported(@NonNull final TokenCommandParame
                     "The min broker protocol version for PopAuthenticationSchemeWithClientKey should be equal or more than 11.0."
                             + " Current required version is set to: " + parameters.getRequiredBrokerProtocolVersion());
         }
+
+        if (parameters.hasNestedAppParameters()) {
+            // Nested app auth is only supported when required protocol version is >= 15
+            if (!StringUtil.isNullOrEmpty(parameters.getChildClientId()) && !StringUtil.isNullOrEmpty(parameters.getChildRedirectUri())
+                    && !BrokerProtocolVersionUtil.canSupportNestedAppAuthentication(requiredProtocolVersion)) {
+                throw new ClientException(ClientException.NESTED_APP_AUTH_NOT_SUPPORTED,
+                        "The min broker protocol version for Nested app auth should be equal or more than 15.0."
+                                + " Current required version is set to: " + parameters.getRequiredBrokerProtocolVersion());
+            } else if ((StringUtil.isNullOrEmpty(parameters.getChildClientId()) || StringUtil.isNullOrEmpty(parameters.getChildRedirectUri()))) {
+                // If only one of the nested app params are sent
+                throw new ClientException(ClientException.NESTED_APP_INVALID_PARAMETERS, "One of the required parameters for nested app is null or empty");
```

Comment:
Can we be specific and tell the customers which one was actually null? We clearly know it